Make `Value.Hash.findFields()` thread-safe with thread-local variables.

blackwinter opened the pull request:
Data structures in `Value.Hash.findFields(String, Set<String>)` are not safe to use across multiple threads. Switch to thread-local variables instead.

Resolves #255.

dr0i commented:
This is *awsome* re performance! Didn't heard before of `ThreadLocal`. Thx very much!

blackwinter commented:
Um, I said I'll merge it tomorrow. Now I have to abort my Limetrans tests :disappointed:

dr0i commented:
Yeah, you said this, but at [another place](https://github.com/metafacture/metafacture-fix/issues/255#issuecomment-1274315320) and I had read it too late. Sry.

blackwinter commented:
It's customary for the committer to merge their own pull requests, isn't it?

dr0i commented:
Depends on the commiter I would say. As I was assigned as Reviewer AND as Assignee I thought this ould imply to do both. But it's fine with me and now I will remember that you will merge your own PRs.

blackwinter commented:
Okay, thanks. I understand assignee to mean "the one responsible for working on the issue/pull request". As reviewer that means you're reviewing, as committer that means I'm addressing your feedback or merging if the pull request was approved.

I've restored the branch for now so the Limetrans tests can continue running. I'll only have to repeat the second master run manually.

dr0i commented:
But you are right, I think:
["The creator of the pull request merges the pull request ..."](https://github.com/metafacture/metafacture-core/blob/master/CONTRIBUTING.md#contributing-code). I will stick to that, reassigning if I am assigned as Assignee.
